clients/src/main/java/org/apache/kafka/clients/admin/KafkaAdminClient.java

```diff
@@ -1402,6 +1407,15 @@ int numPendingCalls() {
         return runnable.pendingCalls.size();
     }
 
+    /**
+     * Fail the given future when a response handler expected a result for an entity but no result was present.
+     * @param future The future to fail.
+     * @param message The message to fail the future with
+     */
+    private void partialResponse(KafkaFutureImpl<?> future, String message) {
```

Comment:
I am not entirely convinced by the name. At the end, the method complete a future with the given message. There is nothing specific to handling partial responses. I would rename it to something more generic or simply remove it as we don't gain much with it.

Reply:
I agree it's not doing much. What value it has is in trying to handle these cases in a consistent way, and being able to more easily discover/reason about the call sites. Maybe something like `invalidBrokerResponse()` would be a better name? But if you don't like that I'm happy to remove it.

Reply:
I see your point. I am definitely for reusing logic as much as possible for those cases. What about going an extra step and do something like this then (code not tested)? Just an idea...

```
public static <K, V> void completeUnrealizedFutures(
            Map<K, KafkaFutureImpl<V>> futures,
            Function<K, Boolean> filter,
            Function<K, String> messageFormatter) {
        for (Map.Entry<K, KafkaFutureImpl<V>> entry : futures.entrySet()) {
            K key = entry.getKey();
            KafkaFutureImpl<V> future = entry.getValue();
            if (!future.isDone() && filter.apply(key)) {
                future.completeExceptionally(new ApiException(messageFormatter.apply(key)));
            }
        }
    }

    public static  <K, V> void completeUnrealizedFutures(
            Map<K, KafkaFutureImpl<V>> futures,
            unction<K, String> messageFormatter) {
        completeUnrealizedFutures(futures, (key) -> true, messageFormatter);
    }
```

It would allow the following:

```
completeUnrealizedFutures(topicFutures,
     replica -> replica.brokerId() == brokerId,
     topic -> "The response from broker " + brokerId + " did not contain a result for replica " + replica);

completeUnrealizedFutures(topicFutures,
     topic -> "The server response did not contain a reference to node " + topic);
```

Reply:
That works for me, especially if this is something you're going to make more use of for the other response handlers in the admin client.